src/figuresClasses.ts

```diff
@@ -1,19 +1,99 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  color: Color;
+
+  a: number;
+
+  b: number;
+
+  c: number;
 
+  constructor(color: Color, a: number, b: number, c: number) {
+    const isValidProps: boolean = a > 0 && b > 0 && c > 0;
+    const isValidSide: boolean = a + b > c && b + c > a && a + c > b;
```

Comment:
typeScript already know about this type
```suggestion
    const isValidProps = a > 0 && b > 0 && c > 0;
    const isValidSide = a + b > c && b + c > a && a + c > b;
```